pkg/depend/write.go

```diff
@@ -27,5 +46,50 @@ func (lock *Lock) WriteLock(node *Node) error {
 		return err
 	}
 
-	return lock.cache.PushNotes(node.url, ref_lock_name)
+	if err = lock.cache.PushNotes(node.url, ref_lock_name); err != nil {
+		return err
+	}
+
+	if err = lock.populateLockRef(node); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (lock *lock) removeLock(node *Node) error {
+	if err := lock.populateLockRef(node); err != nil {
+		return err
+	}
+	if err := lock.cache.RemoveNotes(node.url, ref_lock_name, lock.lockref.object); err != nil {
+		return err
+	}
+	if err := lock.cache.PushNotes(node.url, ref_lock_name); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (lock *lock) populateLockRef(node *Node) error {
+	byte_s, err := lock.cache.ListNotes(node.url, ref_lock_name)
+	if err != nil {
+		return err
+	}
+	// For Windows compatability
+	noteRefs := strings.Split(strings.TrimSpace(strings.ReplaceAll(string(byte_s), "\r\n", "\n")), "\n")
```

Comment:
Is it better to use a regex here?

Reply:
Replaced with `regexp.MustCompile("\r\n|\n").Split(strings.TrimSpace(string(byte_s)), -1)` Not sure that it's that much simpler but the new method doesn't mutate the string, which I think is an overall net benefit.